terraform_json not working

Steps to reproduce:
1. Install checkov: `pip3 install checkov`
2. Download example from tests: https://github.com/bridgecrewio/checkov/blob/main/tests/terraform_json/examples/cdk.tf.json
3. Run checkov: `checkov -f cdk.tf.json --framework terraform_json -o json`

Gives the following output:
```
2023-08-29 12:40:54,038 [MainThread  ] [ERROR]  There are no runners to run. This can happen if you specify a file type and a framework that are not compatible (e.g., `--file xyz.yaml --framework terraform`), or if you specify a framework with missing dependencies (e.g., helm or kustomize, which require those tools to be on your system). Running with LOG_LEVEL=DEBUG may provide more information.
{
    "passed": 0,
    "failed": 0,
    "skipped": 0,
    "parsing_errors": 0,
    "resource_count": 0,
    "checkov_version": "2.4.14"
}
```

What I would expect?
A runner runs with some tests.

<details>
<summary>Debug Output</summary>
LOG_LEVEL=DEBUG checkov -f cdk.tf.json --framework terraform_json
2023-08-29 12:39:24,168 [MainThread  ] [DEBUG]  Leveraging the bundled IAM Definition.
2023-08-29 12:39:24,168 [MainThread  ] [DEBUG]  Leveraging the IAM definition at /usr/local/lib/python3.9/dist-packages/policy_sentry/shared/data/iam-definition.json
2023-08-29 12:39:24,463 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760> with order 1
2023-08-29 12:39:24,463 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,463 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>]
2023-08-29 12:39:24,464 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220> with order 11
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0> with order 0
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0>, <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.fixes_integration.FixesIntegration object at 0x7fc2b7ab4b20> with order 10
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,465 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0>, <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.fixes_integration.FixesIntegration object at 0x7fc2b7ab4b20>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.repo_config_integration.RepoConfigIntegration object at 0x7fc2b7ab4f70> with order 0
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0>, <checkov.common.bridgecrew.integration_features.features.repo_config_integration.RepoConfigIntegration object at 0x7fc2b7ab4f70>, <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.fixes_integration.FixesIntegration object at 0x7fc2b7ab4b20>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.suppressions_integration.SuppressionsIntegration object at 0x7fc2b7ac77c0> with order 2
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,466 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0>, <checkov.common.bridgecrew.integration_features.features.repo_config_integration.RepoConfigIntegration object at 0x7fc2b7ab4f70>, <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.suppressions_integration.SuppressionsIntegration object at 0x7fc2b7ac77c0>, <checkov.common.bridgecrew.integration_features.features.fixes_integration.FixesIntegration object at 0x7fc2b7ab4b20>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,467 [MainThread  ] [DEBUG]  Adding the IntegrationFeatureRegistry <checkov.common.bridgecrew.integration_features.features.licensing_integration.LicensingIntegration object at 0x7fc2b7ac7c10> with order 6
2023-08-29 12:39:24,467 [MainThread  ] [DEBUG]  self.features after the sort:
2023-08-29 12:39:24,467 [MainThread  ] [DEBUG]  [<checkov.common.bridgecrew.integration_features.features.policy_metadata_integration.PolicyMetadataIntegration object at 0x7fc2b7ab45e0>, <checkov.common.bridgecrew.integration_features.features.repo_config_integration.RepoConfigIntegration object at 0x7fc2b7ab4f70>, <checkov.common.bridgecrew.integration_features.features.custom_policies_integration.CustomPoliciesIntegration object at 0x7fc2bd520760>, <checkov.common.bridgecrew.integration_features.features.suppressions_integration.SuppressionsIntegration object at 0x7fc2b7ac77c0>, <checkov.common.bridgecrew.integration_features.features.licensing_integration.LicensingIntegration object at 0x7fc2b7ac7c10>, <checkov.common.bridgecrew.integration_features.features.fixes_integration.FixesIntegration object at 0x7fc2b7ab4b20>, <checkov.common.bridgecrew.integration_features.features.policies_3d_integration.Policies3DIntegration object at 0x7fc2b7a9e220>]
2023-08-29 12:39:24,484 [MainThread  ] [DEBUG]  Loading external checks from /home/node/.local/lib/python3.9/site-packages/checkov/bicep/checks/graph_checks
2023-08-29 12:39:24,484 [MainThread  ] [DEBUG]  Searching through ['__pycache__'] and ['SQLServerAuditingEnabled.json', '__init__.py']
2023-08-29 12:39:24,485 [MainThread  ] [DEBUG]  Searching through [] and ['__init__.cpython-39.pyc']
2023-08-29 12:39:24,574 [MainThread  ] [DEBUG]  Popen(['git', 'version'], cwd=/workspaces/SBC-Infrastructure, universal_newlines=False, shell=None, istream=None)
2023-08-29 12:39:24,579 [MainThread  ] [DEBUG]  Popen(['git', 'version'], cwd=/workspaces/SBC-Infrastructure, universal_newlines=False, shell=None, istream=None)
2023-08-29 12:39:24,773 [MainThread  ] [DEBUG]  No API key present; setting include_all_checkov_policies to True
2023-08-29 12:39:24,777 [MainThread  ] [DEBUG]  Run metadata: {
  "checkov_version": "2.4.14",
  "python_executable": "/usr/bin/python3",
  "python_version": "3.9.2 (default, Feb 28 2021, 17:03:44) \n[GCC 10.2.1 20210110]",
  "checkov_executable": "/usr/local/bin/checkov",
  "args": [
    "Command Line Args:   -f cdk.tf.json --framework terraform_json",
    "Defaults:",
    "  --branch:          master",
    "  --download-external-modules:False",
    "  --external-modules-download-path:.external_modules",
    "  --evaluate-variables:True",
    "  --secrets-scan-file-type:[]",
    "  --block-list-secret-scan:[]",
    "  --summary-position:top",
    "  --mask:            []",
    "  --secrets-history-timeout:12h",
    ""
  ],
  "OS_system_info": "Linux-5.15.90.1-microsoft-standard-WSL2-x86_64-with-glibc2.31",
  "CPU_architecture": "",
  "Python_implementation": "CPython"
}
2023-08-29 12:39:24,777 [MainThread  ] [DEBUG]  Using cert_reqs None
2023-08-29 12:39:24,777 [MainThread  ] [DEBUG]  Successfully set up HTTP manager
2023-08-29 12:39:24,777 [MainThread  ] [DEBUG]  Resultant set of frameworks (removing skipped frameworks): terraform_json
2023-08-29 12:39:24,778 [MainThread  ] [DEBUG]  BC_SOURCE = cli, version = 2.4.14
2023-08-29 12:39:24,778 [MainThread  ] [DEBUG]  terraform_json_runner declares no system dependency checks required.
2023-08-29 12:39:24,778 [MainThread  ] [DEBUG]  No API key found. Scanning locally only.
2023-08-29 12:39:24,851 [MainThread  ] [DEBUG]  Got checkov mappings and guidelines from Bridgecrew platform
2023-08-29 12:39:24,852 [MainThread  ] [DEBUG]  Loading external checks from /home/node/.local/lib/python3.9/site-packages/checkov/terraform_json/checks/graph_checks
2023-08-29 12:39:24,856 [MainThread  ] [DEBUG]  Running without API key, so only open source runners will be enabled
2023-08-29 12:39:24,856 [MainThread  ] [DEBUG]  Filtered list of policies: []
2023-08-29 12:39:24,856 [MainThread  ] [DEBUG]  Received the following policy-level suppressions, that will be skipped from running: []
2023-08-29 12:39:24,856 [MainThread  ] [DEBUG]  Filtered runners based on file type(s). Result: []
2023-08-29 12:39:24,856 [MainThread  ] [ERROR]  There are no runners to run. This can happen if you specify a file type and a framework that are not compatible (e.g., `--file xyz.yaml --framework terraform`), or if you specify a framework with missing dependencies (e.g., helm or kustomize, which require those tools to be on your system). Running with LOG_LEVEL=DEBUG may provide more information.
2023-08-29 12:39:24,856 [MainThread  ] [DEBUG]  Should run contributor metrics report: None

       _               _              
   ___| |__   ___  ___| | _______   __
  / __| '_ \ / _ \/ __| |/ / _ \ \ / /
 | (__| | | |  __/ (__|   < (_) \ V / 
  \___|_| |_|\___|\___|_|\_\___/ \_/  
                                      
By bridgecrew.io | version: 2.4.14 
</details>

--- a/checkov/terraform_json/parser.py
+++ b/checkov/terraform_json/parser.py
@@ -94,8 +94,8 @@ def prepare_definition(definition: dict[str, Any]) -> dict[str, Any]:
             if block_name == COMMENT_FIELD_NAME or block_name in LINE_FIELD_NAMES:
                 continue
 
-            if block_type == BlockType.RESOURCE:
-                # resource have an extra nested level resource_type -> resource_name -> resource_config
+            if block_type in (BlockType.RESOURCE, BlockType.DATA):
+                # data/resource have an extra nested level resource_type -> resource_name -> resource_config
                 for resource_name, resource_config in config.items():
                     if resource_name in IGNORE_FILED_NAMES:
                         continue

--- a/checkov/terraform_json/runner.py
+++ b/checkov/terraform_json/runner.py
@@ -48,7 +48,7 @@ class TerraformJsonRunner(TerraformRunner):
             external_registries=external_registries,
             source=source,
         )
-        self.file_extensions = TF_JSON_POSSIBLE_FILE_ENDINGS  # override what gets set from the TF runner
+        self.file_extensions = (".json",)  # just '.json' not 'tf.json' otherwise it will be filtered out
         self.graph_registry = get_graph_checks_registry(super().check_type)
 
         self.definitions: dict[str, dict[str, Any]] = {}  # type:ignore[assignment]  # need to check, how to support subclass differences

--- a/checkov/terraform_json/runner.py
+++ b/checkov/terraform_json/runner.py
@@ -19,7 +19,7 @@ from checkov.runner_filter import RunnerFilter
 from checkov.terraform.checks.resource.registry import resource_registry
 from checkov.terraform.graph_builder.local_graph import TerraformLocalGraph
 from checkov.terraform.runner import Runner as TerraformRunner
-from checkov.terraform_json.utils import get_scannable_file_paths, TF_JSON_POSSIBLE_FILE_ENDINGS, create_definitions
+from checkov.terraform_json.utils import get_scannable_file_paths, create_definitions
 
 if TYPE_CHECKING:
     from checkov.common.graph.checks_infra.registry import BaseRegistry

--- a/tests/terraform_json/examples/cdk.tf.json
+++ b/tests/terraform_json/examples/cdk.tf.json
@@ -16,6 +16,18 @@
       }
     }
   },
+  "data": {
+    "aws_caller_identity": {
+      "current": {
+        "//": {
+          "metadata": {
+            "path": "AppStack/current",
+            "uniqueId": "current"
+          }
+        }
+      }
+    }
+  },
   "output": {
     "bucket_arn": {
       "value": "${aws_s3_bucket.bucket.arn}"

--- a/tests/terraform_json/test_graph_manager.py
+++ b/tests/terraform_json/test_graph_manager.py
@@ -33,7 +33,7 @@ def test_build_graph_from_definitions(graph_connector):
     )
 
     # then
-    assert len(local_graph.vertices) == 5
+    assert len(local_graph.vertices) == 6
 
     bucket_idx = local_graph.vertices_block_name_map["resource"]["aws_s3_bucket.bucket"][0]
     bucket = local_graph.vertices[bucket_idx]
@@ -42,8 +42,8 @@ def test_build_graph_from_definitions(graph_connector):
     assert bucket.id == "aws_s3_bucket.bucket"
     assert bucket.source == "Terraform"
     assert bucket.attributes[CustomAttributes.RESOURCE_TYPE] == ["aws_s3_bucket"]
-    assert bucket.attributes[START_LINE] == 34
-    assert bucket.attributes[END_LINE] == 53
+    assert bucket.attributes[START_LINE] == 46
+    assert bucket.attributes[END_LINE] == 65
     assert bucket.config == {
         "aws_s3_bucket": {
             "bucket": {
@@ -53,32 +53,25 @@ def test_build_graph_from_definitions(graph_connector):
                             {
                                 "comment": "Access logging not needed",
                                 "id": "CKV_AWS_18",
-                                "__startline__": 38,
-                                "__endline__": 41,
+                                "__startline__": 50,
+                                "__endline__": 53,
                             }
                         ],
-                        "__startline__": 36,
-                        "__endline__": 43,
+                        "__startline__": 48,
+                        "__endline__": 55,
                     },
                     "metadata": {
                         "path": "AppStack/bucket",
                         "uniqueId": "bucket",
-                        "__startline__": 44,
-                        "__endline__": 47,
+                        "__startline__": 56,
+                        "__endline__": 59,
                     },
-                    "__startline__": 35,
-                    "__endline__": 48,
+                    "__startline__": 47,
+                    "__endline__": 60,
                 },
-                "tags": [
-                    {
-                        "Name": "example",
-                        "Private": "true",
-                        "__startline__": 49,
-                        "__endline__": 52,
-                    }
-                ],
-                "__startline__": 34,
-                "__endline__": 53,
+                "tags": [{"Name": "example", "Private": "true", "__startline__": 61, "__endline__": 64}],
+                "__startline__": 46,
+                "__endline__": 65,
                 "__address__": "aws_s3_bucket.bucket",
             }
         }

